File: output/copilot/javascript/results/q4-super-ugly-number.js
/** https://leetcode.com/problems/super-ugly-number/ */
//A super ugly number is a positive integer whose prime factors are in the array primes.
//Given an integer n and an array of integers primes, return the nth super ugly number.
//The nth super ugly number is guaranteed to fit in a 32-bit signed integer.

/**
 * @param {number} n
 * @param {number[]} primes
 * @return {number}
 */
 var nthSuperUglyNumber = function(n, primes) {
    let ugly = [1];
let index = [0];
for(let i = 1; i < n; i++) {
   let min = Number.MAX_SAFE_INTEGER;
   let minIndex = -1;
   for(let j = 0; j < primes.length; j++) {
       let temp = ugly[index[j]] * primes[j];
       if(temp < min) {
           min = temp;
           minIndex = j;
       }
   }
   ugly[i] = min;
   index[minIndex]++;
}
return ugly[n-1]; 
};

File: output/copilot/java/results/q4-super-ugly-number.java
/** https://leetcode.com/problems/super-ugly-number/ */
//A super ugly number is a positive integer whose prime factors are in the array primes.
//Given an integer n and an array of integers primes, return the nth super ugly number.
//The nth super ugly number is guaranteed to fit in a 32-bit signed integer.

class Solution {
    public int nthSuperUglyNumber(int n, int[] primes) {
        int[] ugly = new int[n];
        ugly[0] = 1;
        int[] index = new int[primes.length];
        for (int i = 1; i < n; i++) {
            int min = Integer.MAX_VALUE;
            for (int j = 0; j < primes.length; j++) {
                min = Math.min(min, primes[j] * ugly[index[j]]);
            }
            ugly[i] = min;
            for (int j = 0; j < primes.length; j++) {
                if (primes[j] * ugly[index[j]] == min) {
                    index[j]++;
                }
            }
        }
        return ugly[n - 1];
    }
}